File: remote/src/main/java/com/kelsos/mbrc/dao/ArtistDao.java
package com.kelsos.mbrc.dao;

import com.kelsos.mbrc.RemoteDatabase;
import com.raizlabs.android.dbflow.annotation.Column;
import com.raizlabs.android.dbflow.annotation.PrimaryKey;
import com.raizlabs.android.dbflow.annotation.Table;
import com.raizlabs.android.dbflow.structure.BaseModel;

@Table(database = RemoteDatabase.class, name = "artists") public class ArtistDao extends BaseModel {

  @Column @PrimaryKey(autoincrement = true)private long id;
  @Column private String name;
  @Column(name = "date_added") private long dateAdded;
  @Column(name = "date_updated") private long dateUpdated;
  @Column(name = "date_deleted") private long dateDeleted;

  public String getName() {
    return name;
  }

  public ArtistDao setName(String name) {
    this.name = name;
    return this;
  }

  public long getId() {
    return id;
  }

  public ArtistDao setId(long id) {
    this.id = id;
    return this;
  }

  public long getDateAdded() {
    return dateAdded;
  }

  public void setDateAdded(long dateAdded) {
    this.dateAdded = dateAdded;
  }

  public long getDateUpdated() {
    return dateUpdated;
  }

  public void setDateUpdated(long dateUpdated) {
    this.dateUpdated = dateUpdated;
  }

  public long getDateDeleted() {
    return dateDeleted;
  }

  public void setDateDeleted(long dateDeleted) {
    this.dateDeleted = dateDeleted;
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/data/dao/ArtistDao.kt
package com.kelsos.mbrc.data.dao

import com.kelsos.mbrc.data.RemoteDatabase
import com.raizlabs.android.dbflow.annotation.Column
import com.raizlabs.android.dbflow.annotation.Index
import com.raizlabs.android.dbflow.annotation.IndexGroup
import com.raizlabs.android.dbflow.annotation.PrimaryKey
import com.raizlabs.android.dbflow.annotation.Table
import com.raizlabs.android.dbflow.kotlinextensions.modelAdapter
import com.raizlabs.android.dbflow.structure.Model

@Table(database = RemoteDatabase::class,
    name = "artists",
    indexGroups = arrayOf(IndexGroup(number = 1, name = "artist_name_index")))
data class ArtistDao(
    @Column @Index(indexGroups = intArrayOf(1)) var name: String = "",
    @Column(name = "date_added") var dateAdded: Long = 0,
    @Column(name = "date_updated") var dateUpdated: Long = 0,
    @Column(name = "date_deleted") var dateDeleted: Long = 0,
    @Column @PrimaryKey(autoincrement = true) var id: Long = 0) : Model {

  override fun load() = modelAdapter<ArtistDao>().load(this)

  override fun insert(): Long = modelAdapter<ArtistDao>().insert(this)

  override fun save() = modelAdapter<ArtistDao>().save(this)

  override fun update() = modelAdapter<ArtistDao>().update(this)

  override fun exists(): Boolean = modelAdapter<ArtistDao>().exists(this)

  override fun delete() = modelAdapter<ArtistDao>().delete(this)
}
